### ISSUE TITLE
[BUG] Wrong namespace is used when both datetime and date is used, causing a PydanticSchemaGenerationError

### ISSUE THREAD

**Issue by emreozcan** [kmbhm1/supabase-pydantic#48]
**Describe the bug**
When both `date` and `timestamp`s are used in a schema, these get translated into Python classes as `datetime.date` and `datetime`. Because supabase-pydantic adds the import statement as `from datetime import date, datetime`, the `datetime.date` is interpreted as the `date` method of the `datetime` class, and not as the `date` member of the module `datetime`.

**To Reproduce**
1. Execute this SQL:
	```sql
	create table testing (id serial primary key, a timestamp, b date);
	```
2. Execute this in your shell:
	```sh
	sb-pydantic gen \
	    --type pydantic \
	    --framework fastapi\
	    --dir uniiq_backend/entities \
	    --local
	```
3. Line 3 will read: `from datetime import date, datetime`
4. The appropriate line will read: `b: datetime.date`
5. Importing the generated module will raise the following exception:
	```
	pydantic.errors.PydanticSchemaGenerationError: Unable to generate pydantic-core schema for <method 'date' of 'datetime.datetime' objects>. Set `arbitrary_types_allowed=True` in the model_config to ignore this error or implement `__get_pydantic_core_schema__` on your type to fully support it.
	```

**Expected behavior**
The annotation should be `b: date` and not `b: datetime.date`

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: Windows 10 Build 19045.4894
 - Browser N/A
 - Version N/A

**Additional context**
- supabase-pydantic: 0.14.5
- pydantic: 2.9.2

**Comment by emreozcan** [comment]
I am willing to contribute a PR if you would be interested!
(but I will need a few days to get to it)

**Comment by kmbhm1** [comment]
@emreozcan  Sorry for the delay. I've been rather busy & will look into this over the next several days. More to come.

**Comment by kmbhm1** [comment]
@emreozcan Sorry for the delay on this - can you try [0.14.6?](https://pypi.org/project/supabase-pydantic/0.14.6/) and confirm?

**Comment by emreozcan** [comment]
Can confirm it is resolved on 0.14.6. Thanks a lot!
